constraint/pkg/client/errors.go

```diff
@@ -58,3 +59,27 @@ func (e *MissingTemplateError) Error() string {
 func NewMissingTemplateError(mapKey string) error {
 	return &MissingTemplateError{mapKey}
 }
+
+// Errors is a list of error.
+//
+// Deprecated: Use a structured result type if it is important to disambiguate
```

Comment:
Does this mean have individual custom types for each error?

Reply:
It's very situational. If we want to report individual error codes, then we should use a structured error type. If we need to programmatically work with information the error can communicate, then the error should have that structure. In general we want to treat errors as black-box as possible unless we have a functional reason to do otherwise.